### ISSUE TITLE
Improve pytest developer experience

### ISSUE THREAD

**Issue by dmos62** [centerofci/mathesar#1368]
## Problem

- Our pytest test suites take too long to run. The `db/tests/` suite alone takes 15 minutes to run on my machine (my i5-4670K is dated, but not outdated). This degrades ability to quickly iterate.
- Also, our tests are brittle, in that whether they pass might depend on the order in which they are run. Them being order-dependent makes it hard to run tests out-of-order, which is often preferable (`--ff` option runs failed and unrun tests first).
- Further, our fixtures are bloated and difficult to understand. Because of this, a developer will spend more time writing his tests (or editing others') and the result will be worse.
- Futher, our test coverage is inefficient due to heavy overlap, meaning that many aspects of Mathesar are tested many times over. This also contributes to brittleness, since one change might affect many tests.

## Proposed solution
- To reduce runtime: refactor fixtures to allow for running multiple tests in parallel;
- To address order-dependence: setup pytest to run tests in random order by default;
- To improve readability: establish a coherent fixture system;
- To improve overlap: identify most expensive tests and factor out overlapping coverage.

## Additional context
Issue related to test coverage overlap: https://github.com/centerofci/mathesar/issues/1138

**Comment by kgodey** [comment]
@dmos62 Is this fixed by your refactor?

**Comment by kgodey** [comment]
There is a separate issue for connection management: #592 

@dmos62 Can you open separate issues for each of the other three items so that we can work on them separately?

**Comment by dmos62** [comment]
Split this issue into four issues (see mentioned-in issues above).
